lighthouse/datadog_checks/lighthouse/lighthouse.py

```diff
@@ -0,0 +1,82 @@
+# the following try/except block will make the custom check compatible with any Agent version
+try:
+    # first, try to import the base class from old versions of the Agent...
+    from checks import AgentCheck
+except ImportError:
+    # ...if the above failed, the check is running in Agent version 6 or later
+    from datadog_checks.checks import AgentCheck
+    from datadog_checks.errors import CheckException
+
+from datadog_checks.utils.subprocess_output import get_subprocess_output
+import json
+
+# content of the special variable __version__ will be shown in the Agent status page
+__version__ = "1.0.0"
+
+
+class LighthouseCheck(AgentCheck):
+    def check(self, instance):
+        EXPECTED_RESPONSE_CODE = "NO_ERROR"
+        lighthouse_url = instance.get('url')
+        lighthouse_name = instance.get('name')
+
+        if not lighthouse_url or not lighthouse_name:
+            self.log.error("missing instance url or name")
+            raise CheckException("missing lighthouse instance url or name, please fix yaml")
+
+        cmd = ["/usr/local/bin/lighthouse",
+               lighthouse_url,
+               "--output",
+               "json",
+               "--quiet",
+               "--chrome-flags='--headless'"]
+
+        json_string, error_message, exit_code = self._get_lighthouse_report(cmd, self.log, False)
+
+        # check for error since we have raise_on_empty_output set to False
+        if exit_code > 0:
+            self.log.error("lighthouse subprocess error {0} exit code {1} for url: {2}"
+                           .format(error_message, exit_code, lighthouse_url)
+                           )
+            raise CheckException(json_string, error_message, exit_code)
+
+        try:
+            data = json.loads(json_string)
+        except Exception as e:
+            self.log.warn("lighthouse response JSON structure different than expected for url: %s" % lighthouse_url)
+            raise CheckException(error_message, exit_code, e)
+
+        if data["runtimeError"]["code"] == EXPECTED_RESPONSE_CODE:
+            score_accessibility = round(data["categories"]["accessibility"]["score"] * 100)
+            score_best_practices = round(data["categories"]["best-practices"]["score"] * 100)
+            score_performance = round(data["categories"]["performance"]["score"] * 100)
+            score_pwa = round(data["categories"]["pwa"]["score"] * 100)
+            score_seo = round(data["categories"]["seo"]["score"] * 100)
+        else:
+            err_code = data["runtimeError"]["code"]
+            err_msg = data["runtimeError"]["message"]
+            self.log.warn("not collecting lighthouse metrics for url {0} runtimeError code {1} message {2}"
+                          .format(lighthouse_url, err_code, err_msg)
+                          )
+            return
+        # add tags
+        try:
+            tags = instance['tags']
```

Comment:
If `tags` is `None` (users have not defined any custom tags) this will throw a `NoneType` Exception. You can use `instance.get('tags', [])` instead (empty list as default value). Example of this in he disk check: https://github.com/DataDog/integrations-core/blob/master/disk/datadog_checks/disk/disk.py#L56.
Then I don't think you'd need to check if `tags` is a list.

Reply:
updated to .get('tags',[]) but i think i ought to keep the check if it's a list in case the user has formatted the config incorrectly, just to be defensive?

Reply:
We don't do it in other checks as far as I can see, but I don't think it hurts to check that indeed.